packages/sources/view-function/src/endpoint/function.ts

```diff
@@ -0,0 +1,64 @@
+import { AdapterError, Requester, Validator } from '@chainlink/ea-bootstrap'
+import { ExecuteWithConfig, InputParameters } from '@chainlink/types'
+import { Config } from '../config'
+import { Contract, utils } from 'ethers'
+
+export const supportedEndpoints = ['function']
+
+export const inputParameters: InputParameters = {
+  signature: ['function', 'signature'],
+  address: ['address', 'contract'],
+  inputParams: false,
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
+  const validator = new Validator(request, inputParameters)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.id
+  const address = validator.validated.data.address
+  const fnSignature = validator.validated.data.signature
+  const params = validator.validated.data.inputParams || []
+
+  if (typeof address !== 'string' || !address.length) {
+    throw new AdapterError({
+      jobRunID,
+      message: `Input, at 'address' or 'contract' path, must be a non-empty string.`,
+      statusCode: 400,
+    })
+  }
+
+  if (typeof fnSignature !== 'string' || !fnSignature.length) {
+    throw new AdapterError({
+      jobRunID,
+      message: `Input, at 'signature' or 'function' path, must be a non-empty string.`,
+      statusCode: 400,
+    })
+  }
```

Comment:
Isn't this already handled by the validator?

Reply:
Validator throws error only if the field is `undefined`. `null` and `empty strings` are not validated

Reply:
Created a ticket for it: https://app.shortcut.com/chainlinklabs/story/19390/ea-validator-should-check-if-request-param-strings-are-empty